File: js/lib/MessageBroker/MessageBroker.js
class MessageBroker {
  constructor(dispatcher) {
    this.connectionRetries = 10;
    this.connectionInterval = 2000;

    this.dispatcher = dispatcher;
  }

  async connect() {
    try {
      await this.dispatcher.connect({
        connectionInterval: this.connectionInterval,
        connectionRetries: this.connectionRetries
      })
    } catch (error) {

      // TODO: Create an application error

      throw new Error("MessageBroker connection refused");
    }
  }

  createTopic({ name, routing }) {
    let topic = this.dispatcher.createTopic({ name, routing });

    return topic;
  }
}

module.exports = MessageBroker

File: ruby/lib/MessageBroker/MessageBroker.rb
require_relative 'errors/MessageBrokerConnectionRefused'

class MessageBroker
  def initialize dispatcher:
    @CONNECTION_RETRIES = 10
    @CONNECTION_INTERVAL = 2

    @dispatcher = dispatcher 
  end   

  def connect
    begin
      @dispatcher.connect(
        connectionInterval: @CONNECTION_INTERVAL, 
        connectionRetries: @CONNECTION_RETRIES
      )
    rescue DispatcherConnectionRefused
      raise MessageBrokerConnectionRefused
    end  
  end

  def createTopic name:, routing:
    topic = @dispatcher.createTopic(name: name, routing: routing)

    return topic
  end  
end